tests/e2e/scripts/modules/install.sh

```diff
@@ -199,42 +199,80 @@ verify_installation() {
   
   log "Verifying CSI driver installation in namespace: $namespace..."
   
-  # Wait for the pods to be running
-  log "Waiting for CSI driver pods to be in Running state in namespace: $namespace..."
+  # Wait for the pods to be running and ready
+  log "Waiting for CSI driver pods to be ready in namespace: $namespace..."
   
   # Maximum wait time in seconds (5 minutes)
-  MAX_WAIT_TIME=300
-  WAIT_INTERVAL=10
-  ELAPSED_TIME=0
+  local TIMEOUT=300s
   
-  while [ $ELAPSED_TIME -lt $MAX_WAIT_TIME ]; do
-    if exec_cmd kubectl get pods -n $namespace | grep -q "Running"; then
-      log "CSI driver pods are now running in namespace: $namespace."
-      
-      exec_cmd kubectl get pods -n $namespace
+  # Check if CSI driver is registered first (which should happen immediately after Helm install)
+  log "Checking if CSI driver is registered..."
+  
+  if ! exec_cmd kubectl get csidrivers | grep -q "s3.csi.scality.com"; then
+    error "CSI driver is not registered properly. Installation may have failed."
+    exec_cmd kubectl get csidrivers
+    return
+  fi
+  
+  log "CSI driver is registered successfully."
+  
+  # Get all CSI related pods - using a broader selector that matches the Scality helm chart's labels
+  local csi_pod_name_pattern="s3-csi"
+  
+  # Wait for the pods to be created first
+  WAIT_COUNT=0
+  MAX_WAIT=30
+  WAIT_INTERVAL=5
+  
+  # Wait up to 2.5 minutes just for pods to be created (image pulls can take time)
+  log "Waiting for CSI driver pods to be created..."
+  while [ $WAIT_COUNT -lt $MAX_WAIT ]; do
+    # Check if there are any CSI driver pods with names containing s3-csi
+    if [ "$(kubectl get pods -n $namespace | grep "${csi_pod_name_pattern}" 2>/dev/null | wc -l)" -gt 0 ]; then
```

Comment:
I see some pipes, don't you want to have at the top of the script something like `set -euo pipefail`